cilium-cli/connectivity/builder/to_fqdns.go

```diff
@@ -12,45 +12,54 @@ import (
 	"github.com/cilium/cilium/cilium-cli/utils/features"
 )
 
+// Common expectations function for FQDN tests
+func fqdnsExpectations(ct *check.ConnectivityTest) func(a *check.Action) (egress, ingress check.Result) {
+	return func(a *check.Action) (egress, ingress check.Result) {
+		if a.Destination().Address(features.IPFamilyAny) == ct.Params().ExternalOtherTarget {
+			if a.Destination().Path() == "/" || a.Destination().Path() == "" {
+				// Expect packets to other external target to be dropped.
+				return check.ResultDropCurlTimeout, check.ResultNone
+			}
+			// Else expect HTTP drop by proxy
+			return check.ResultDNSOKDropCurlHTTPError, check.ResultNone
+		}
+
+		extTarget := ct.Params().ExternalTarget
+		if a.Destination().Port() == 80 && a.Destination().Address(features.GetIPFamily(extTarget)) == extTarget {
+			if a.Destination().Path() == "/" || a.Destination().Path() == "" {
+				egress = check.ResultDNSOK
+				egress.HTTP = check.HTTP{
+					Method: "GET",
+					// Trim the trailing dot, if any, to match the behavior of the curl
+					// action and make sure that flow validation can succeed.
+					URL: fmt.Sprintf("http://%s/", strings.TrimSuffix(extTarget, ".")),
+				}
+				return egress, check.ResultNone
+			}
+			// Else expect HTTP drop by proxy
+			return check.ResultDNSOKDropCurlHTTPError, check.ResultNone
+		}
+		// No HTTP proxy on other ports
+		return check.ResultDNSOKDropCurlTimeout, check.ResultNone
+	}
+}
+
 type toFqdns struct{}
 
 func (t toFqdns) build(ct *check.ConnectivityTest, templates map[string]string) {
-	// This policy only allows port 80 to domain-name, default one.one.one.one., DNS proxy enabled.
+	if !ct.Params().ExternalTargetIPv6Capable {
```

Comment:
Let's add a brief comment here  explaining why the test is skipped in this case.

Reply:
Incidentally, does this mean that if someone deploys the test in a single-stack IPv4 environment, the test will be skipped?

Reply:
> We can skip the test, but ideally it should check for L7 proxy and skip accordingly. 

Yeah, this would be my counter-proposal.

> If the test is meant for IPv6 FQDN without proxy, then expectations and CNP deployment should be updated accordingly. [Still not clear if that was the initial intention]

I don't understand what it means to test FQDN without proxy. FQDN policy as a feature relies on the proxy to pull the DNS information for implementation.

Reply:
I agree. From my perspective, it's pretty odd to configure --enable-proxy=false. I see that flag as more for historic reasons due to lack of support on ARM64 years ago, but I think it's not ideal as a user-facing flag. It's not related to the functionality users want, but instead an architectural detail. Cilium will by default for instance avoid running Envoy if there's no L7 rules. Maybe we could improve the UX around that overall, but the reality today is that the flag exists and it's possible to disable it. So yes I agree we can aim to just detect and handle that case in CLI for these tests.